### PR TITLE
Adding omitempty support for struct Marshalling

### PR DESCRIPTION
**Pull request by jakdept**
I've modified the encoder to honor a tag of:

```
`xmlrpc:"tag,omitempty"`
```

This allows someone to skip sending values that are empty - as defined by `reflect.Value.IsZero()`.

Note, this may be a breaking change? If anyone has a struct with a tag of `something,omitempty` this will also strip `,omitempty` from the end of that tag and possibly not send it? I would expect this would be unintended behavior?

I also added a test to explicitly show what happens when you encode a empty struct:

```xml
<value><struct></struct></value>
```

I don't know if that's intended, and I didn't see anything in the XMLRPC spec about an empty `<struct>` element, so I don't know if that's right?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Second argument of NewClient function is an object that implements
 [http.RoundTripper](http://golang.org/pkg/net/http/#RoundTripper)
 interface, it can be used to get more control over connection options.
 By default it initialized by http.DefaultTransport object.
+These tags support the omitempty property.
```
Comment (icholy) [inline README.md]: This doesn't seem like the right place for this comment.

File: encoder.go [hunk 2]
```diff
@@ -90,21 +91,26 @@ func encodeStruct(val reflect.Value) ([]byte, error) {
 
 	t := val.Type()
 	for i := 0; i < t.NumField(); i++ {
-		b.WriteString("<member>")
 		f := t.Field(i)
 
 		name := f.Tag.Get("xmlrpc")
+		// if the tag has the omitempty property, skip it
+		if strings.HasSuffix(name, ",omitempty") && val.FieldByName(f.Name).IsZero() {
```
Comment (icholy) [inline encoder.go]: You can do `val.Field(i).IsZero()` here. Also, assign the result of `val.Field(i)` to variable and re-use it in the `encodeValue` call.

**Comment by icholy** [comment]
What's the exact use-case here? Maybe I'm just lucky, but I've never seen a xmlrpc endpoint with optional parameters.

**Comment by jakdept** [comment]
You hit it exactly on the head - I'm working against an API that has optional parameters for some calls.

In this case, the `methodRoute` is always the same, the first `param` is always an authentication object, the second and third `param` are actually the routing keys, and the fourth param is the data - sometimes a struct, sometimes a single non-optional value.

Specifying the value in the request as 0 if an integer, or a empty string if a string, is obviously different than not specifying it.

File: README.md (hunk 1 above)
Comment (jakdept) [reply README.md]: I'm all for changing it to wherever seems more appropriate. tbh, I had written the change, then realized I did not mention it in the readme.

File: encoder.go (hunk 2 above)
Comment (jakdept) [reply encoder.go]: `encodeValue()` later was already using `val.FieldByName()`, so I reused what was there to avoid stepping on toes.

I've saved and reused it. I also went ahead and renamed the other related elements of the struct for clarity.

File: README.md (hunk 1 above)
Comment (icholy) [reply README.md]: Move it under the "Structs decoded to struct by following rules:" heading. Also fix that heading to say "Structs encoded to struct by following rules:"

**Comment by icholy** [comment]
@jakdept thanks for the contribution.
